### PR TITLE
Use fetch instead of apiFetch.

### PR DESCRIPTION
**Pull request by ankitrox**
## Summary

Addresses issue:

- #9485 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.
- [ ] Ensure there are no unexpected significant changes to file sizes.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/hooks/useMonitorInternetConnection.js [hunk 1]
```diff
@@ -53,15 +54,9 @@ export function useMonitorInternetConnection() {
...
 
 		try {
-			const connectionCheckResponse = await apiFetch( {
-				path: '/google-site-kit/v1/',
-			} );
-
-			// We are only interested if the request was successful, to
-			// confirm online status.
-			const canReachConnectionCheck = !! connectionCheckResponse;
+			await fetch( `${ rootURL }google-site-kit/v1/` );
```
Comment (aaemnnosttv) [inline assets/js/hooks/useMonitorInternetConnection.js:57]: Let's use `URL` to combine the relative path with the base safely. From what I've seen it seems to always include a trailing slash, but it is a filterable value so better to not assume.

```suggestion
			await fetch( new URL( '/google-site-kit/v1/', rootURL ) );
```

File: assets/js/hooks/useMonitorInternetConnection.test.js [hunk 2]
```diff
@@ -40,7 +42,7 @@ describe( 'useMonitorInternetConnection', () => {
 		} );
 	};
 
-	const connectionCheckEndpoint = '/google-site-kit/v1/?_locale=user';
+	const connectionCheckEndpoint = `${ rootURL }google-site-kit/v1/`;
```
Comment (aaemnnosttv) [inline assets/js/hooks/useMonitorInternetConnection.test.js:45]: Same here

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @ankitrox – this looks very close, just one suggestion about the way the URL is constructed.

**Comment by ankitrox** [comment]
Thanks @aaemnnosttv for reviewing the PR and providing the feedback.

I've made the suggested changes and assigning it to you for another review.

File: assets/js/hooks/useMonitorInternetConnection.js [hunk 3]
```diff
@@ -53,15 +54,9 @@ export function useMonitorInternetConnection() {
...
 
 		try {
-			const connectionCheckResponse = await apiFetch( {
-				path: '/google-site-kit/v1/',
-			} );
-
-			// We are only interested if the request was successful, to
-			// confirm online status.
-			const canReachConnectionCheck = !! connectionCheckResponse;
+			await fetch( new URL( '/google-site-kit/v1/', rootURL ) );
```
Comment (aaemnnosttv) [inline assets/js/hooks/useMonitorInternetConnection.js:57]: After some more thought here, I think this is not quite the right solution (criticizing my own IB here! 😄).

The fact that we're reaching for some internal api-fetch data here is a smell that we should be using that instead. The reason I suggested using `fetch` instead was to avoid the additional cases where `api-fetch` throws, where `fetch` only throws when the request fails. Essentially we should be able to do the same using `api-fetch` with an additional check in the throw block as `api-fetch` returns a specific error in this case. See https://github.com/WordPress/gutenberg/blob/d9b726b8451746703cc1b9680487e3726ab4a03f/packages/api-fetch/src/index.js#L130-L131

In summary, we can keep most of what we had before, but adjusted to use an approach closer to the suggested one here in letting the try/catch determine the result. The difference in using `api-fetch` is that we need to set the online state conditionally depending on the error code if it throws. I.e. if it throws and the error code is `fetch_error`, only then should we set it to be offline. Any other error would mean the request happened, which is all we are concerned about here, and can be ignored.

**Review by aaemnnosttv — COMMENTED** [review]
Thanks @ankitrox – after some reconsideration, I think we should slightly revise the approach to avoid reaching for the API fetch data.

File: assets/js/hooks/useMonitorInternetConnection.js (hunk 3 above)
Comment (ankitrox) [reply assets/js/hooks/useMonitorInternetConnection.js]: Thanks @aaemnnosttv , I've made the changes according to the suggestion.

File: assets/js/hooks/useMonitorInternetConnection.js [hunk 4]
```diff
@@ -63,7 +63,12 @@ export function useMonitorInternetConnection() {
```
Comment (aaemnnosttv) [inline assets/js/hooks/useMonitorInternetConnection.js:63]: We can remove all of the response checking above because we're only concerned with it throwing or not.

**Review by aaemnnosttv — COMMENTED** [review]
Thanks @ankitrox – this can be further simplified a little bit which I'll do now before merging, otherwise LGTM!

**Review by aaemnnosttv — APPROVED** [review]

**Comment by aaemnnosttv** [comment]
E2E failures are unrelated. See #8229
